js/libs/utils.js

```
@@ -0,0 +1,275 @@
+}
+
+/**
+ * Deep merge objects (similar to _.assignIn but recursive)
+ * Replacement for _.assignIn() with deep merging
+ * @param {Object} target - Target object
+ * @param {...Object} sources - Source objects to merge
+ * @returns {Object} Merged object
+ */
+function assignIn(target, ...sources) {
+  if (!target || typeof target !== 'object') {
+    target = {};
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)